File: (7 kyu) Last/(7 kyu) Last.js
// #1
// function last(list) {
//   if (arguments.length > 1) {
//     return arguments[arguments.length - 1];
//   } else if (list.length > 1) {
//     return list[list.length - 1];
//   } else {
//     return list;
//   }
// }
// #2
function last(list) {
  return arguments.length > 1
    ? arguments[arguments.length - 1]
    : list[list.length - 1] || arguments[arguments.length - 1];
}
// #3
function last(list) {
  const last = arguments[arguments.length - 1];
  return last[last.length - 1] || last;
}

File: (7 kyu) Last/(7 kyu) Last.java
import java.util.List;

public class Solution {
  public static <T> T last(final List<T> list) {
    if(list == null || list.isEmpty()) {
      return null;
    } else {
      return list.get(list.size()-1);
    }
  }
  public static char last(final String string) {
    if(string == null || string.isEmpty()) {
      return 0;
    } else {
      return string.charAt(string.length()-1);
    }
  }
  public static <T> T last(final T... list) {
    if(list == null) {
      return null;
    } else {
      return list[list.length-1];
    }
  }  
}
